invoices/invoiceregistry.go

```diff
@@ -396,6 +406,8 @@ func (i *InvoiceRegistry) AddInvoice(invoice *channeldb.Invoice,
 		return 0, err
 	}
 
+	i.expiryWatcher.AddInvoice(paymentHash, invoice)
```

Comment:
What about checking the status of an invoice within `NotifyExitHopHtlc`? Otherwise, it's possible that concurrent with an invoice's expiry to wake up the main watcher loop, an invoice arrives (past expiration), which is then settled. Let's not lose sight of [_fundamental_ issue](https://github.com/lightningnetwork/lnd/issues/3448) we're trying to address here. 
>

Reply:
To me, it seems that the worst that can happen is that you end up with a settled invoice in the database that has a `settle date` that is a fraction of a second after the expiry. It would indeed be better to make it so that this is never possible, but I I don't think it is important enough to address in this PR.